src/SFML/Window/Context.cpp

```diff
@@ -49,19 +49,37 @@ namespace sf
 Context::Context() : m_context(priv::GlContext::create())
 {
     if (!setActive(true))
-    {
         err() << "Failed to set context as active during construction" << std::endl;
-    }
 }
 
 
 ////////////////////////////////////////////////////////////
 Context::~Context()
 {
-    if (!setActive(false))
-    {
+    if (m_context && !setActive(false))
         err() << "Failed to set context as inactive during destruction" << std::endl;
-    }
+}
+
+
+////////////////////////////////////////////////////////////
+Context::Context(Context&& context) noexcept : m_context(std::move(context.m_context))
+{
+    if (!setActive(&context == ContextImpl::currentContext))
+        err() << "Failed to set context as active during move construction" << std::endl;
```

Comment:
Because `setActive` requires `m_context` not to be null, the following code would crash:
```c++
sf::Context c1;
sf::Context c2 = std::move(c1);
sf::Context c3 = std::move(c1);
```
Maybe we can ignore this problem because it is generally bad to use a variable after move, but on the other hand it is simple to fix with the following code:

```suggestion
    if (&context == ContextImpl::currentContext)
        ContextImpl::currentContext = this;
```

Same applies to move assignment operator.

Reply:
I did actually think of that but my impression is that moving out of `c1` twice is something we can assume users aren't allowed to do and thus we don't have to guard against it.

Reply:
Perhaps we `assert` against moving out of an empty context so that this is a hard failure and we prevent UB in Debug builds?

Reply:
I added asserts to protect against this (mis)use of contexts. If someone later shows a valid use case for double-moving out of a context then we can remove the asserts and instead add an `if (m_context)` check to define this behavior in release builds.

Reply:
Apparently we are required to let users move out of the same object many times so we’re forced to add that pointer check back.